src/main/java/org/jabref/gui/customentrytypes/CustomEntryTypeDialogViewModel.java

```diff
@@ -205,10 +205,14 @@ public void apply() {
             List<Field> multilineFieldsForType = allFields.stream().map(FieldViewModel::getField).filter(Field::isMultiLineDefined).collect(Collectors.toList());
             multilineFields.addAll(multilineFieldsForType);
 
-            List<OrFields> requiredFields = allFields.stream().filter(field -> field.getFieldType() == FieldType.REQUIRED).map(FieldViewModel::getField).map(OrFields::new).collect(Collectors.toList());
-            List<BibField> otherFields = allFields.stream().filter(field -> field.getFieldType() == FieldType.OPTIONAL).map(bibField -> new BibField(bibField.getField(), bibField.getFieldPriority())).collect(Collectors.toList());
-
-            BibEntryType newType = new BibEntryType(type.getType(), otherFields, requiredFields);
+            List<OrFields> required = allFields.stream()
+                                               .filter(field -> field.getFieldType() == FieldType.REQUIRED)
+                                               .map(FieldViewModel::getField)
+                                               .map(OrFields::new)
+                                               .collect(Collectors.toList());
+            List<BibField> fields = allFields.stream().map(FieldViewModel::toBibField).collect(Collectors.toList());
```

Comment:
I don't understand this change. Now you are essentially duplicating the required fields?

Reply:
This was the bug i mentioned before. The BibEntryType constructor needs all its fields as BibFields and of that the required fields as OrFields, because BibEntryType, the serializer and parser work handles the fields as 

![grafik](https://user-images.githubusercontent.com/50491877/232137596-cf8991ec-2616-4502-bb6b-2ca442f4e632.png)

![grafik](https://user-images.githubusercontent.com/50491877/232137474-a5e7fd19-ee87-4a09-ba67-ef38bdc7f779.png)

That happens, when someone does not create proper documentation for misunderstandable methods. :smirk: 

![grafik](https://user-images.githubusercontent.com/50491877/232137991-5b4ca964-df9e-4431-a2d6-2c6a836d0627.png)

I fixed that too. :grin: 

![grafik](https://user-images.githubusercontent.com/50491877/232138200-4fb6c262-362a-447e-89c8-6ce4a5dd1d2b.png)

Reply:
Ah yes, now I see 😇